src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineOption.java

```diff
@@ -110,7 +112,10 @@ public boolean isInRange(int positionInLine) {
 		if(getStartPositionInLine() <= characterPosition && characterPosition <= getStartPositionInLine() + optionName.length()) {
 			String description = CamelKModelineOptionNames.getDescription(optionName);
 			if(description != null) {
-				return CompletableFuture.completedFuture(new Hover(Collections.singletonList((Either.forLeft(description)))));
+				Hover hover = new Hover();
+				hover.setContents(Collections.singletonList((Either.forLeft(description))));
+				hover.setRange(new Range(new Position(getLine(), getStartPositionInLine()), new Position(getLine(), getStartPositionInLine() + optionName.length())));
+				return CompletableFuture.completedFuture(hover);
```

Comment:
special case because the model is not using a specific class for the Camel K modeline option name